src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -118,6 +125,7 @@ public function __construct(
         $this->translator = $translator;
         $this->hookManager = $hookManager ?? new HookManager();
         $this->configuration = $configuration ?? new Configuration();
+        $this->embeddedAttributesFilter = $embeddedAttributesFilter;
```

Comment:
```suggestion
        $this->embeddedAttributesFilter = $embeddedAttributesFilter ?? new EmbeddedAttributesFilter();
```
You can use the same principle as what's done for configuration and hook manager, especially since the PHPDoc tends to say that it can not be null, besides you won't need to check the presence of the field in the `getEmbeddedAttributes` function

Reply:
You're right 👍